### PR TITLE
Add BDS process watchdog

### PR DESCRIPTION
**Pull request by bennydiamond**
Can be toggled On/Off in config json
Set to retry 3 times when enabled
Deems process stable when receiving "Server Started" string from BDS. Otherwise, will stop retry if max retry count is reached (hardcoded to 3)

Moved strings coming from BDS into centralized location because I hate string duplication.

### CODE REVIEW

File: src/Automation/ProcessManager.cs [hunk 1]
```diff
@@ -215,6 +241,26 @@ private void OutputTextReceived(object sender, DataReceivedEventArgs e)
...
+        private void ProcessExited(object sender, EventArgs e)
+        {
+            Console.WriteLine("BDS process unexpectedly exited");
+            this.Close();
+            if(++this._failRetryCount < this._maxFailRetryCount)
+            {
+                Console.WriteLine("Retry #" + this._failRetryCount + " to start BDS server process");
+                Start();
+            }
+            else if(this._maxFailRetryCount > 0)
```
Comment (clarkx86) [inline src/Automation/ProcessManager.cs:254]: Shouldn't this be
```csharp
else if._failRetryCount > 0)
```
:)?

**Comment by bennydiamond** [comment]
Closing, I am missing some stuff related to playerCount in Program.cs and I don't have the time to fix it right away. Will re-submit in a few hours.

**Comment by clarkx86** [comment]
Oh, currently making some adjustment and was about to merge :). But no worries, thanks again so far! :)

**Comment by bennydiamond** [comment]
> Oh, currently making some adjustment and was about to merge :). But no worries, thanks again so far! :)

Will merge your changes and submit again once it's ready. Should be in around 4-5 hours.

**Pull request closed without merging**